resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/DelayedSnapshotJob.java

```diff
@@ -56,4 +61,31 @@ public IStatus run(IProgressMonitor monitor) {
 			saveManager.snapshotRequested = false;
 		}
 	}
+
+	/**
+	 * Suspend automatic snapshots until {@link #resume()} is called.
+	 *
+	 * @see #resume()
+	 */
+	void suspend() {
+		suspended.incrementAndGet();
+	}
+
+	/**
+	 * Resume automatic snapshots. This method does not schedule this Job.
+	 *
+	 * @see #suspend()
+	 */
+	void resume() {
+		if (suspended.decrementAndGet() < 0) {
+			suspended.set(0);
+		}
+	}
```

Comment:
Agreed. Sorry about that, I didn't consider the semantics of JobManager.suspend / resume. I was more thinking about the usage pattern where a boolean was stored to call / not call resume again. I think that's highly confusing and prone to invalid state in case multiple threads use these methods.

Reply:
Good point, I haven't thought about the behavior of `suspend` / `resume` in the `JobManager`, only about nesting calls like the ones the `prepare...`/`begin...`/`endOperation` in `Workspace`.